base/Dockerfile

```diff
@@ -52,8 +52,8 @@ USER gitpod
 # use sudo so that user does not get sudo usage info on (the first) login
 RUN sudo echo "Running 'sudo' for Gitpod: success" && \
     # create .bashrc.d folder and source it in the bashrc
-    mkdir /home/gitpod/.bashrc.d && \
-    (echo; echo "for i in \$(ls \$HOME/.bashrc.d/*); do source \$i; done"; echo) >> /home/gitpod/.bashrc
+    mkdir -p /home/gitpod/.bashrc.d && \
+    (echo; echo -e "if ! [ -z \$(ls -A -- \$HOME/.bashrc.d/); then\nfor i in \$(ls -A \$HOME/.bashrc.d/); do source \$HOME/.bashrc.d/\$i; done\nfi"; echo) >> /home/gitpod/.bashrc
```

Comment:
The opening `[` of the if statement is unclosed? 

If using `ls -A` you can skip the if test and go directly to the for loop, because `ls -A` will come out empty if the folder is empty already. 

So you may simplify to:

```shell
(echo; echo "for i in \$(ls -A \$HOME/.bashrc.d/); do source \$HOME/.bashrc.d/\$i; done"; echo) >> /home/gitpod/.bashrc
```

Manual test:

```shell
docker run -it --rm gitpod/workspace-base

# remove the last two lines (the current failing code)
gitpod ~ $ sed -i '$ d'  /home/gitpod/.bashrc
gitpod ~ $ sed -i '$ d'  /home/gitpod/.bashrc

# Now add the new code:
gitpod ~ $ (echo; echo "for i in \$(ls -A \$HOME/.bashrc.d/); do source \$HOME/.bashrc.d/\$i; done"; echo) >> /home/gitpod/.bashrc

# verify result
gitpod ~ $ tail -2  /home/gitpod/.bashrc
for i in $(ls -A $HOME/.bashrc.d/); do source $HOME/.bashrc.d/$i; done

# enter a new shell, see no error appears now
gitpod ~ $ bash
gitpod ~ $ exit

# now create a file an see it is sourced
gitpod ~ $ echo "echo 'I was sourced'" > $HOME/.bashrc.d/10_test.sh
gitpod ~ $ bash
I was sourced
gitpod ~ $ exit
```



```

Reply:
Thanks for the comments.